internal/toolexec/utils/utils.go

```diff
@@ -0,0 +1,35 @@
+package utils
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// ExitIfError calls os.Exit(1) if err is not nil
+func ExitIfError(err error) {
+	if err != nil {
+		log.Printf("%v", err)
+		os.Exit(1)
+	}
+}
+
+// GoBuild builds in provided dir and returns the work directory's true path
+func GoBuild(dir string, args ...string) (string, error) {
+	args = append([]string{"build", "-work", "-a", "-p", "1"}, args...)
```

Comment:
The documentation doesn't tell the whole story here...

- Implicitly says `-work`, but perhaps should mention you're responsible for cleaning up after it unless you wanna keep files around, still
- Doesn't mention it passes `-a`, and MAYBE this should be part of `args...` if the caller wants it
- Doesn't mention it passes `-p 1`, which I'd argue is not "the right way" to do it if there are concurrency issues

Reply:
Right, the `-p` stuff should actually be removed - it's legacy from the bash POC.
I'll enrich the doc.